crates/language/src/language_registry.rs

```diff
@@ -730,21 +727,44 @@ impl LanguageRegistry {
             })
         });
         self.find_matching_language(move |language_name, config, current_best_match| {
+            let filename = filename?;
+
+            let matching_suffix_len = |acc: usize, suffix: &String| {
+                let ext = ".".to_string() + suffix;
+                if filename.ends_with(&ext) || filename == suffix {
+                    acc.max(suffix.len())
+                } else {
+                    acc
+                }
+            };
+
             let path_matches_default_suffix = || {
-                config
-                    .path_suffixes
-                    .iter()
-                    .any(|suffix| path_suffixes.contains(&Some(suffix.as_str())))
```

Comment:
I think you should keep the structure that was there before, but store the length of the matching suffix.

Reply:
OK, done.